build_projects/dotnet-host-build/CompileTargets.cs

```diff
@@ -322,6 +323,107 @@ public static BuildTargetResult CompileCoreHost(BuildTargetContext c)
             return c.Success();
         }
 
+        
+        [Target]
+        [BuildPlatforms(BuildPlatform.Windows)]
+        public static BuildTargetResult CompileUwpHosts(BuildTargetContext c)
+        {
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                return c.Failed();
+
+            var configuration = c.BuildContext.Get<string>("Configuration");
+            string platform = c.BuildContext.Get<string>("Platform");
+
+            // cmake generated build file location artifacts\<config>\uwp\cmake
+            var cmakeOut = Path.Combine(Dirs.Uwp, "cmake");
+            Rmdir(cmakeOut);
+            Mkdirp(cmakeOut);
+
+            // cmake generated build file location artifacts\<config>\uwp
+            var cmakeInstallPrefixDir = Dirs.Uwp;
+
+            // Run the build
+            string uwpSrcDir = Path.Combine(c.BuildContext.BuildDirectory, "src", "uwp");
+
+            // Create .rc files
+            var resourceDir = GenerateVersionResource(c);
+
+            if (configuration.Equals("Release"))
+            {
+                // Cmake calls it "RelWithDebInfo" in the generated MSBuild
+                configuration = "RelWithDebInfo";
+            }
+
+            List<string> cmakeArgList = new List<string>();
+
+            string visualStudio, archMacro, arch;
+            string cmakeResourceDir = $"-DCLI_CMAKE_RESOURCE_DIR:STRING={resourceDir}";
+            string cmakeInstallPrefix = $"-DCMAKE_INSTALL_PREFIX:STRING={cmakeInstallPrefixDir}";
+            string cmakeExtraArgs = null;
+
+            switch (platform.ToLower())
+            {
+                case "x86":
+                    visualStudio = "Visual Studio 14 2015";
+                    archMacro = "-DCLI_CMAKE_PLATFORM_ARCH_I386=1";
+                    cmakeExtraArgs = "-DCMAKE_SYSTEM_VERSION=10.0";
```

Comment:
ExtraArgs are consistent for all architectures and can be moved out instead of repeating them.